File: КОМПОНЕНТЫ/OPartner/Examples/Delphi/exappen1.pas
unit ExAppen1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, OpShared, OpWrd2k, OpWord;

type
  TForm1 = class(TForm)
    btnNewDocument: TButton;
    btnAppendDocument: TButton;
    OpenDialog1: TOpenDialog;
    OpWord1: TOpWord;
    procedure btnNewDocumentClick(Sender: TObject);
    procedure btnAppendDocumentClick(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;
  MainDoc: TOpWordDocument;
implementation

{$R *.DFM}

procedure TForm1.btnNewDocumentClick(Sender: TObject);
begin
  OpWord1.Visible:= True;
  MainDoc:=  OpWord1.Documents.Add;
  MainDoc.Insert('New Document' + #10#13#10#13, True);
end;

procedure TForm1.btnAppendDocumentClick(Sender: TObject);
var
  EndOfDocVariant: OleVariant;
  EndOfDocRange: Range;
begin
  if OpenDialog1.Execute then
  begin
    
    MainDoc.Insert(OpenDialog1.FileName + #10#13#10#13, True);
    EndOfDocVariant:= (MainDoc.AsDocument).Characters.Count - 1;
    EndOfDocRange:= (MainDoc.AsDocument).Range(EndOfDocVariant, EndOfDocVariant);
    EndOfDocRange.InsertFile(OpenDialog1.FileName, emptyParam, emptyParam, emptyParam, emptyParam);
  end;
end;

procedure TForm1.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  if Assigned(MainDoc) then
    MainDoc.Free;
  OpWord1.Connected := False;
end;

procedure TForm1.FormCreate(Sender: TObject);
begin
  OpWord1.Connected := True;
  OpWord1.Visible := False;
end;

end.

File: КОМПОНЕНТЫ/OPartner/Examples/CBuildr5/exappen1.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "ExAppen1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "OpShared"
#pragma link "OpWord"
#pragma link "OpWrd2k"
#pragma resource "*.dfm"

#define spIDocument Opwrd2k::_di__Document  //safe pointer
#define spIRange Opwrd2k::_di_Range         //safe pointer

TForm1 *Form1;
TOpWordDocument *MainDoc;

//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
    : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnNewDocumentClick(TObject *Sender)
{
  OpWord1->Visible = true;
  MainDoc = OpWord1->Documents->Add();
  MainDoc->Insert("New Document\n\n", true);
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnAppendDocumentClick(TObject *Sender)
{
  if (OpenDialog1->Execute()) {
    MainDoc->Insert(OpenDialog1->FileName, true);
    spIDocument IDocument = MainDoc->AsDocument;
    OleVariant EndOfDoc = IDocument->Characters->Count - 1;
    spIRange IRange = IDocument->Content;
    IRange->InsertFile(OpenDialog1->FileName, EmptyParam,
                       EmptyParam, EmptyParam, EmptyParam);
  }
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormActivate(TObject *Sender)
{
  OpWord1->Connected = true;
  OpWord1->Visible = false;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormClose(TObject *Sender, TCloseAction &Action)
{
  if (MainDoc)
    delete MainDoc;
  OpWord1->Connected = false;
}
//---------------------------------------------------------------------------
